core/muon-config.js

```diff
@@ -31,17 +32,78 @@ var RQ = require("async-rq");
 
 var MuonConfig = function () {};
 
-MuonConfig.prototype.generateMuon = function() {
+MuonConfig.prototype.generateMuon = function(serviceName, discoveryUrl) {
     logger.debug("Initialising Muon from Environmental information, checking ...");
+     logger.debug("Configuring via function args: discoveryUrl=" + discoveryUrl + ", serviceName=" + serviceName);
+
+    var config = {
+             "serviceName": undefined,
+             "tags" : [ "" ],
+             "discovery": {
+               "type": "amqp",
+               "url": undefined
+             },
+             "transports": [
+               { "type":"amqp", "url": undefined }
+             ]
+    }
+
+
+    var muonConfigUrlEnv = process.env.MUON_CONFIG_URL;
+    var muonConfigServiceNameEnv = process.env.MUON_CONFIG_SVC;
+
+
+
+    if (discoveryUrl && serviceName) {
+        logger.debug("Configuring via function args: discoveryUrl=" + discoveryUrl + ", serviceName=" + serviceName);
+         config.discovery.url = discoveryUrl;
+         config.transports[0].url = discoveryUrl;
+         config.serviceName = serviceName;
+    } else if (muonConfigUrlEnv && muonConfigServiceNameEnv) {
+            logger.debug("Configuring muon via shell: $MUON_CONFIG_URL=" + envVar + ", $MUON_CONFIG_SVC=" + muonConfigServiceName);
+            config.discovery.url = muonConfigUrlEnv;
+            config.transports[0].url = muonConfigUrlEnv;
```

Comment:
We must remember this for when we add another transport. This locks it to a single one